### ISSUE TITLE
require-hook: false positive on `it.aroundAll()` and other member-form hooks

### ISSUE THREAD

**Issue by williamthorsen** [vitest-dev/eslint-plugin-vitest#955]
## Description

Vitest 4 exposes every lifecycle hook as a member of the test API (the `Hooks` interface in `@vitest/runner`: `test.beforeAll`, `test.aroundAll`, `test.aroundEach`, and the rest), and calling a hook on an extended test, e.g. `const it = test.extend(...); it.aroundAll(...)`, is how a suite gets typed access to its scoped fixtures. #912 added support for the bare `aroundAll`/`aroundEach` imports, but no member-form hook chain exists in `ValidVitestFnCallChains`, so `parseVitestFnCall` fails to recognize `it.aroundAll(...)` and `require-hook` reports the hook registration itself as code that should be inside a hook.

## Reproduction

`eslint.config.mjs`:

```js
import vitest from '@vitest/eslint-plugin';

export default [
  {
    files: ['**/*.test.js'],
    plugins: { vitest },
    rules: { 'vitest/require-hook': 'warn' },
  },
];
```

`around-all-member.test.js`:

```js
import { test } from 'vitest';

const it = test.extend({
  fixture: async ({}, use) => {
    await use('hello');
  },
});

it.aroundAll(async (runSuite) => {
  await runSuite();
});

it('uses the fixture', ({ fixture, expect }) => {
  expect(fixture).toBe('hello');
});
```

Observed:

```
9:1  warning  This should be done within a hook  vitest/require-hook
```

Expected: no warning; `it.aroundAll(...)` is a hook registration.

## Root cause

`src/utils/valid-vitest-fn-call-chains.ts` contains the bare hook names, and the `HookName` enum already includes `aroundAll` and `aroundEach`, but the table has no `test.<hook>` / `it.<hook>` chain for any hook. The parser resolves the extended `it` back to the `test` import, builds the chain `test.aroundAll`, misses the table, and returns `null`, so `require-hook` treats the statement as arbitrary top-level code. Every rule that relies on `parseVitestFnCall` to recognize hooks shares this blind spot for the member forms. User-side workaround: `'vitest/require-hook': ['warn', { allowedFunctionCalls: ['it.aroundAll'] }]`.

## Environment

- `@vitest/eslint-plugin`: 1.6.27
- `eslint`: 10.8.1
- `vitest`: 4.1.10

Related: #912 (added the bare forms), #914 (another missing chain in the same table).
